src/Airship/Internal/Helpers.hs

```diff
@@ -22,12 +23,25 @@ import qualified Network.HTTP.Types        as HTTP
 import qualified Network.Wai               as Wai
 import           Network.Wai.Parse
 import           System.Random
+import           Text.Read
 
+import           Airship.Headers
 import           Airship.Internal.Decision
 import           Airship.Internal.Route
 import           Airship.Resource
 import           Airship.Types
-import           Airship.Headers
+
+-- | Reads the parameter of the provided name and attempt to convert
+-- it to the desired 'Read'-implementing type. If the parameter is not
+-- found or cannot be read successfully, the provided handler is invoked
+-- (this may often be `halt status404`).
+readParam :: Read a => Text -> Handler s m a -> Handler s m a
```

Comment:
Hi. Sorry to butt in, just started using airship at work and keen to get involved if/where I can.

Just out of curiosity, what is the expected use-case for `readParam`? In particular would it be used for `String` or `Text`? I ask because I've seen people burnt by the `Read` instance of both which expects the "quotes", but compiles quite happily. I could imagine that half the time you just want to lookup a string param, in which case this function isn't going to help. Just my 2c. :)

Unrelated, FWIW I agree that it should be a `400` status.